OpenRA.Game/Graphics/RgbaColorRenderer.cs

```diff
@@ -0,0 +1,254 @@
+				var prevCorner = width / 2 * new float2(-prevDir.Y, prevDir.X);
+				ca = IntersectionOf(start - prevCorner, prevDir, start - corner, dir);
+				cb = IntersectionOf(start + prevCorner, prevDir, start + corner, dir);
+			}
+
+			var limit = closed ? points.Length : points.Length - 1;
+			for (var i = 0; i < limit; i++)
+			{
+				var next = points[(i + 2) % points.Length];
+				var nextDir = (next - end) / (next - end).Length;
+				var nextCorner = width / 2 * new float2(-nextDir.Y, nextDir.X);
+
+				// Vertices for the corners joining start-end to end-next
+				var cc = closed || i < limit ? IntersectionOf(end + corner, dir, end + nextCorner, nextDir) : end + corner;
+				var cd = closed || i < limit ? IntersectionOf(end - corner, dir, end - nextCorner, nextDir) : end - corner;
+
+				// Fill segment
+				if (nv + 4 > renderer.TempBufferSize)
+					Flush();
+
+				vertices[nv++] = new Vertex(ca + Offset, r, g, b, a);
+				vertices[nv++] = new Vertex(cb + Offset, r, g, b, a);
+				vertices[nv++] = new Vertex(cc + Offset, r, g, b, a);
+				vertices[nv++] = new Vertex(cd + Offset, r, g, b, a);
+
+				// Advance line segment
+				end = next;
+				dir = nextDir;
+				corner = nextCorner;
+
+				ca = cd;
+				cb = cc;
+			}
+		}
+
+		public void DrawLine(float2[] points, float width, Color color, bool connectSegments = false)
+		{
+			if (!connectSegments)
+			{
+				if (points.Length < 2)
+					return;
+
+				for (var i = 1; i < points.Length; i++)
+					DrawLine(points[i - 1], points[i], width, color);
+			}
+			else
+				DrawConnectedLine(points, width, color, false);
+		}
+
+		public void DrawPolygon(float2[] vertices, float width, Color color)
+		{
+			DrawConnectedLine(vertices, width, color, true);
+		}
+
+		public void DrawRect(float2 tl, float2 br, float width, Color color)
+		{
+			var tr = new float2(br.X, tl.Y);
+			var bl = new float2(tl.X, br.Y);
+			DrawPolygon(new[] { tl, tr, br, bl }, width, color);
+		}
+
+		public void FillRect(float2 tl, float2 br, Color color)
+		{
+			var tr = new float2(br.X, tl.Y);
+			var bl = new float2(tl.X, br.Y);
+			FillRect(tl, tr, br, bl, color);
+		}
+
+		public void FillRect(float2 a, float2 b, float2 c, float2 d, Color color)
+		{
+			renderer.CurrentBatchRenderer = this;
+
+			if (nv + 4 > renderer.TempBufferSize)
+				Flush();
+
+			color = Util.PremultiplyAlpha(color);
+			var cr = color.R / 255.0f;
+			var cg = color.G / 255.0f;
+			var cb = color.B / 255.0f;
+			var ca = color.A / 255.0f;
+
+			vertices[nv++] = new Vertex(a + Offset, cr, cg, cb, ca);
+			vertices[nv++] = new Vertex(b + Offset, cr, cg, cb, ca);
+			vertices[nv++] = new Vertex(c + Offset, cr, cg, cb, ca);
+			vertices[nv++] = new Vertex(d + Offset, cr, cg, cb, ca);
+		}
+
+		public void FillEllipse(RectangleF r, Color color, int vertices = 32)
+		{
+			// TODO: Create an ellipse polygon instead
+			var a = (r.Right - r.Left) / 2;
+			var b = (r.Bottom - r.Top) / 2;
+			var xc = (r.Right + r.Left) / 2;
```

Comment:
What do the `c` suffixes indicate here?
If that is "change" can then become `d` for delta (consistency)?